### PR TITLE
Split state and record for a list item, fixed #129

### LINKED ISSUE #129: Would be nice if ember-frost-list worked with EmberData objects as the selected items
This mostly works except for range-selection. Range-selection fails because an Ember Data object doesn't have a `findIndex()` method. 

https://cl.ly/2M2y0h2b1D3h

If `ember-frost-list` used a different method here, it might be possible to simply pass Ember Data objects directly. Instead, a dev has to do a `selectedObjects.map((item) => item.toJSON({includeId: true})` to convert the objects to POJOs.

### PR DESCRIPTION
**Pull request by dafortin**
### This project uses [semver](semver.org), please check the scope of this pr:

- [ ] #none# - documentation fixes and/or test additions
- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG
* **Split** states and record for a list item
* **Fixed** #129 
* **Changed** version of `smoke-and-mirrors`
* **Added** blueprint check

### CODE REVIEW

**Comment by dafortin** [comment]
I think it's a major since I change a few interface for the components. Let me know if you are not agreeing. 

Also feel free to propose different names for the interface attributes.

**Comment by sglanzer-fire** [comment]
We need to do everything possible avoid #major# releases - can you summarize the breaking changes?

**Comment by dafortin** [comment]
I think we can probably do a #patch# since the interfaces changes are for components that I'm not expecting people to use on their own.

File: addon/components/frost-list.js [hunk 1]
```diff
@@ -129,17 +129,6 @@ export default Component.extend({
 
   // == Lifecycle Hooks =======================================================
 
-  didUpdateAttrs ({newAttrs}) {
```
Comment (sglanzer-fire) [inline addon/components/frost-list.js]: Sorry - can you put a quick explanation on the removal of this?

File: addon/templates/components/frost-list-content-container.hbs [hunk 2]
```diff
@@ -1,44 +1,31 @@
...
+  alwaysUseDefaultHeight=alwaysUseDefaultHeight
+  bufferSize=bufferSize
+  content=items
+  defaultHeight=defaultHeight
+  firstReached=onLoadPrevious
+  lastReached=onLoadNext
+  scrollPosition=scrollTop
+  useContentProxy=false
 
-{{#if pagination}}
```
Comment (sglanzer-fire) [inline addon/templates/components/frost-list-content-container.hbs]: Is it necessary to remove the "basic" {{#each}} loop in the pagination case?  I'd prefer to not introduce {{#vertical-collection}} for those people using the pagination scenario in a minor release

File: addon/templates/components/frost-list-content-container.hbs [hunk 3]
```diff
@@ -1,44 +1,31 @@
...
-    didMountCollection=null
-    idForFirstItem=null
-    renderFromLast=false
-    renderAllInitially=false
-    shouldRender=true
-    minimumMovement=15
-    containerSelector=null
-    containerHeight=null
+  key='@identity'
+  resizeDebounce=64
```
Comment (sglanzer-fire) [inline addon/templates/components/frost-list-content-container.hbs]: 64?

**Comment by sglanzer-fire** [comment]
Looks good overall, just a few comments

File: addon/components/frost-list.js (hunk 1 above)
Comment (dafortin) [reply addon/components/frost-list.js]: We don't need the frost scroll with the `vertical-collection`. The vertical collection has the scrolling capability.

File: addon/templates/components/frost-list-content-container.hbs (hunk 2 above)
Comment (dafortin) [reply addon/templates/components/frost-list-content-container.hbs]: I can remove it. I thought you asked me to replace the `each` by a vertical-collection.

File: addon/templates/components/frost-list-content-container.hbs (hunk 3 above)
Comment (dafortin) [reply addon/templates/components/frost-list-content-container.hbs]: Not sure what this is. I simply used the same vertical collection as below... I feel like we have a lot of properties that might not be necessary in the vertical collection.

Comment (sglanzer-fire) [reply addon/templates/components/frost-list-content-container.hbs]: Ok, that's fine - it's like 15 FPSish - just an odd default, but no problem

File: addon/templates/components/frost-list-content-container.hbs (hunk 2 above)
Comment (sglanzer-fire) [reply addon/templates/components/frost-list-content-container.hbs]: Yeah, I'd like to keep the existing pagination so that we don't risk breaking

Comment (sglanzer-fire) [reply addon/templates/components/frost-list-content-container.hbs]: Ping me when that's done and I'll merge

Comment (dafortin) [reply addon/templates/components/frost-list-content-container.hbs]: Sure I reverted this :)

File: addon/components/frost-list.js (hunk 1 above)
Comment (dafortin) [reply addon/components/frost-list.js]: I reverted this 👍

Comment (sglanzer-fire) [reply addon/components/frost-list.js]: Did you push the changeset?  It still appears as removed in the PR

File: addon/templates/components/frost-list-content-container.hbs (hunk 2 above)
Comment (sglanzer-fire) [reply addon/templates/components/frost-list-content-container.hbs]: Same here, still appears as removed in the PR

**Comment by dafortin** [comment]
@sglanzer pushed the changes

**Comment by sglanzer-fire** [comment]
Approved!

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-list/pull-request/130/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-list)
